modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/frontend/data/set/view/table/ObjectEntriesTableFDSView.java

```diff
@@ -38,94 +51,231 @@
+				fdsTableSchemaField.setContentRenderer("boolean");
 			}
-			else {
-				fdsTableSchemaField =
-					fdsTableSchemaBuilder.addFDSTableSchemaField(
-						fieldName, objectField.getLabel(locale, true));
+		}
 
-				if (Objects.equals(objectField.getDBType(), "Boolean")) {
-					fdsTableSchemaField.setContentRenderer("boolean");
-				}
-			}
+		if (Validator.isNotNull(contentRenderer)) {
+			fdsTableSchemaField.setContentRenderer(contentRenderer);
+		}
 
-			fdsTableSchemaBuilder.addFDSTableSchemaField(fdsTableSchemaField);
+		if (!Objects.equals(type, "Blob") && sortable) {
+			fdsTableSchemaField.setSortable(true);
+		}
 
-			if (!Objects.equals(objectField.getDBType(), "Blob") &&
-				objectField.isIndexed()) {
+		fdsTableSchemaBuilder.addFDSTableSchemaField(fdsTableSchemaField);
+	}
 
-				fdsTableSchemaField.setSortable(true);
-			}
+	private void _addNonobjectField(
+		FDSTableSchemaBuilder fdsTableSchemaBuilder, String objectFieldName) {
+
+		if (Objects.equals(objectFieldName, "creator")) {
+			_addFDSTableSchemaField(
+				null, fdsTableSchemaBuilder, "creator.name", "author", false,
+				null);
+		}
+		else if (Objects.equals(objectFieldName, "dateCreated")) {
+			_addFDSTableSchemaField(
+				null, fdsTableSchemaBuilder, "dateCreated", "created-date",
+				false, "Date");
 		}
+		else if (Objects.equals(objectFieldName, "dateModified")) {
+			_addFDSTableSchemaField(
+				null, fdsTableSchemaBuilder, "dateModified", "modified-date",
+				false, "Date");
+		}
+		else if (Objects.equals(objectFieldName, "id")) {
+			_addFDSTableSchemaField(
+				"actionLink", fdsTableSchemaBuilder, "id", "id", false, null);
+		}
+		else if (Objects.equals(objectFieldName, "status")) {
+			_addFDSTableSchemaField(
+				"status", fdsTableSchemaBuilder, "status", "status", false,
+				null);
+		}
+	}
 
-		FDSTableSchemaField statusFDSTableSchemaField =
-			fdsTableSchemaBuilder.addFDSTableSchemaField("status", "status");
+	private void _addObjectField(
+		FDSTableSchemaBuilder fdsTableSchemaBuilder, Locale locale,
+		ObjectField objectField) {
 
-		statusFDSTableSchemaField.setContentRenderer("status");
+		if (!_ffObjectViewConfigurationActivator.enabled() &&
+			Validator.isNotNull(objectField.getRelationshipType())) {
 
-		fdsTableSchemaBuilder.addFDSTableSchemaField("creator.name", "author");
+			return;
+		}
 
-		return fdsTableSchemaBuilder.build();
+		if (Validator.isNull(objectField.getRelationshipType())) {
+			_addFDSTableSchemaField(
+				null, fdsTableSchemaBuilder,
+				_getFieldName(
+					objectField.getListTypeDefinitionId(),
+					objectField.getName()),
+				objectField.getLabel(locale, true), objectField.isIndexed(),
+				objectField.getDBType());
+		}
+		else if (Objects.equals(
+					objectField.getRelationshipType(),
+					ObjectRelationshipConstants.TYPE_ONE_TO_MANY)) {
+
+			String objectFieldName = objectField.getName();
+
+			String sanitizedObjectFieldName = objectFieldName.substring(
+				0, objectFieldName.length() - 2);
+
```

Comment:
You got it! 😉

Reply:
This will provoke a series of inlinings. Do you agree with them?
```
_addFDSTableSchemaField(
	null, fdsTableSchemaBuilder,
	_getFieldName(
		objectField.getListTypeDefinitionId(),
		StringBundler.concat(
			StringUtil.replaceLast(objectField.getName(), "Id", ""),
			StringPool.PERIOD, titleObjectField.getName())),
	objectField.getLabel(locale, true), false,
	titleObjectField.getDBType());
```

Reply:
@rodrigopaulino It's ok for me too